src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -71,6 +71,12 @@ public function getEnvPlaceholders()
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
```

Comment:
@nicolas-grekas Can you maybe explain the reason for the change here a bit more, I'm not seeing it. 
With `$this->envPlaceholders += $newPlaceholders;` you're saying merge the new into the existing, fair enough. Then it's looping through each of the new ones and merging the existing ones with it? Won't this already have been done by the first merge?